tests/e2e/tests/content-type-builder/collection-type/create-collection-type.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { test, expect } from '@playwright/test';
+import { login } from '../../../utils/login';
+import { resetDatabaseAndImportDataFromPath, resetFiles } from '../../../scripts/dts-import';
+import { waitForRestart } from '../../../utils/restart';
+
+test.describe('Create collection type', () => {
+  test.beforeEach(async ({ page }) => {
+    await resetFiles();
+    await resetDatabaseAndImportDataFromPath('with-admin.tar');
+    await page.goto('/admin');
+    await login({ page });
+
+    await page.getByRole('link', { name: 'Content-Type Builder' }).click();
+
+    // close the tutorial modal if it's visible
+    const modal = page.getByRole('button', { name: 'Close' });
+    if (modal.isVisible()) {
+      await modal.click();
+      await expect(modal).not.toBeVisible();
+    }
+  });
+
+  // TODO: each test should have a beforeAll that does this, maybe combine all the setup into one util to simplify it
+  // to keep other suites that don't modify files from needing to reset files, clean up after ourselves at the end
+  test.afterAll(async () => {
+    await resetFiles();
```

Comment:
why are you doing it afterAll but also in `beforeEach`? Maybe it should be in `afterEach` instead?

Reply:
It's a tricky question here.

In this file, the correct thing would be only the `beforeEach` (since each test makes changes) and then other suites would ensure on their own it was clean with a beforeAll/beforeEach as needed.  But doing the filesystem reset is slow and only a couple suites make file changes (ctb and uploading), and I didn't want to update every other test file to do that.

So I put the "bad" code (trying to clean up for other tests in afterAll is bad practice because you're not guaranteed it will run) separate even though it means we run it one extra time, because ultimately I want to remove it altogether once we have a faster method to detect/reset the files that would be ok to add universally.